### DISCUSSION TITLE
View and play the waveform from right to left?

### DISCUSSION THREAD

**Discussion by Binah-CS** [katspaugh/wavesurfer.js#3579] [Q&A]
When working with RTL languages, it would be very helpful to play a voice wave from right to left.
How can this be implemented with wavesurfer - with all the methods working correctly (regions, timeline, minimap, etc)?

**Comment by katspaugh** [comment]
Unfortunately, v7 doesn't support this.  V6 did – https://wavesurfer.xyz/example/rtl/. Not sure about plugins though.

It's easy enough to fip the waveform with CSS (`transform: scaleX(-1)`) but mouse interactions probably assume left-to-right.

**Comment by Binah-CS** [comment]
So... if I use V6 - it will work?

**Comment by omerman** [comment]
Is an RTL support planned on v7? 😢

**Comment by entonbiba** [comment]
@Binah-CS @omerman for now you can try using the css method by doing the following. It would probably need a bit of tweaking for markers and stuff but works great.

```
<style>
        #waveform {
            transform: scaleX(-1);
        }
        #waveform-controls {
            transform: scaleX(-1);
        }
        .marker-label {
            transform: scaleX(-1);
        }
</style>
```

**Reply by entonbiba** [reply]
#3729

**Comment by Binah-CS** [comment]
I'll try. We do have markers, time stamps and markups, so we need to make
sure it all flips accordingly. We also have long files, so we need the
scrolling to work correctly as well.
